src/query.ts

```diff
@@ -507,6 +510,45 @@ export class Query<T extends Model> {
     return _.pick(instance, Object.keys(getModelAssociations(this.model))) as T;
   }
 
+/**
+ * Returns the associations that correlate to the current include options
+ * @param {boolean} required? Whether or not to only get required associations (associations that are not nullable).
+ * @return {any[]} An array of association descriptors.
+ */
+  private getIncludedAssociations(required?: boolean): any[] {
+    return (this.options.includes || [])
+      .map(include => this.internalModel['associations'][include.as])
+      .filter(assoc => assoc && assoc.foreignKeyAttribute && !assoc.foreignKeyAttribute.allowNull);
+  }
+
+/**
+ * Updates the instance with any associations that are required for creation.
+ * @param instance The model instance to associate.
+ * @param assocValues The associations of the instance, as a plain object.
+ */
+  private prepareRequiredAssociations(instance: T, assocValues: {}): void {
```

Comment:
Can you please merge this into prepareAssociations instead of it being separate?
Also preferably have it not mutate the value but rather pick the relevant values using lodash/whatever.